### ISSUE TITLE
Choral Explanations

### ISSUE THREAD

**Issue by techsmyth** [alkem-io/alkemio#936]
## Description
Engaging the community to build a shared understanding, from multiple perspectives, on key questions. The format in which people would answer is markdown, you may want to add documents or a video to support your answer.

This is similar to the Stack Overflow Q&A format. 

This makes for a lower barrier to entry for participants than for example a greenfield wiki or document management system would have. Also the modality encourages community members to engage personally, in their own voice, and potentially with multiple modalities.

https://hapgood.us/2016/05/13/choral-explanations/

## Initiative / goal
Engaging the community

### Hypothesis
Surfacing up and being able to learn from the collective wisdom of a community is key to making progress, and to getting an engaged community that finds value in the collaboration.

## Acceptance criteria and must have scope
- [ ] Comments on Callouts
- [ ] Add liking to Profiles and thus to cards
- [ ] Different lay-out for the cards

Later:
- [ ] Filtering of Callouts as they scale 
- [ ] Vote a contribution up or down
- [ ] Easy way to link to other contributions, for example the ability to search for contributions in the Hub to link such as issues to other issues in Github

## Stakeholders
TIP
Alkemio

## Timeline
tbd

## Design
Figma file:
_Comments are highly appreciated. When commenting in Figma, please also leave a note in the comments box below this issue to notify us_

**Comment by techsmyth** [comment]
@Fintokyo

**Comment by sander** [comment]
Thanks for the work on this so far @techsmyth and @SimoneZaza. The Knowledge Base UX has improved since our initial requests in 2022. Consequently, the [TIP knowledge base](https://alkem.io/tip/knowledge-base) is used by some more participants. However, I notice that:

- even our most motivated community members keep getting frustrated by the current Knowlegde Base UX;
- from the “outer circle” within our initiative, I keep getting feedback that the Knowledge Base is too difficult for themselves or their colleagues, even for reading.

Update to the changes requested in [December 2022](https://github.com/alkem-io/alkemio/discussions/913):

1. ~~Raising the comment limit from its current 500 characters.~~ ✅ Solved by using callout answers.
2. ~~Enable full data export for a callout or for a full Hub using an API. A potential low-effort implementation could be to enable users to create a personal access token (like on GitHub) to call the GraphQL API directly. Alternatively, since you are on the Ory stack for IAM, I can share my experiences using Ory Hydra for OAuth 2.0. Tracking issue: [Authenticated command line clients #247](https://github.com/alkem-io/alkemio/issues/247)~~ ✅ Solved by making the [TIP knowledge base](https://alkem.io/tip/knowledge-base) public.
3. **Ability to edit comments while keeping version history.** Ideally I can authorize a group of community members to edit other people’s comments as well. ⏳ This request now applies to callout answers. Are these editable by their original authors? Can others review the version history?
4. **Render the full title of a card instead of just the part that fits on a single line.** ⏳ Feedback still applicable, but lower priority given the use of callouts.
5. ~~Support uploading images to cards and comments. Tracking issue:~~ https://github.com/alkem-io/alkemio/issues/667 ✅ Solved.
6. ~~Support embedding images and videos to encourage sharing knowledge using various media.~~ ✅ Solved at least for images. For videos our users apply YouTube for now.
7. **Ability to select an explicit Creative Commons license for a callout** (in our case, CC BY 4.0 or CC0 1.0), requiring contributors to provide their contributions under this license. Make this explicit in the content submission form. This enables reuse of the knowledge in this knowledge base. ⚠️ This is still not fixed. The current UX is painful. Note that the request now applies to both callouts and callout responses.
8. **Ability to upvote cards and comments**, such as on Stack Overflow. ⏳ This is still not fixed. The request now applies to callout and callout responses.
9. ~~Ability to respond to a comment. I see no need for more levels of threading, as this can easily become confusing. Tracking issue:~~ [Comments: Threads + replies #871](https://github.com/alkem-io/alkemio/issues/871) ✅ Not relevant anymore since our switch to callouts.
10. ~~Remove cover photo placeholders on cards, in order to not “punish” contributions which have no visual material.~~ ✅ Fixed.
11. ~~Enable reordering callouts on the Explore tab. Tracking issue:~~ [Callouts: roadmap!! subscriptions, ordering, states etc #758](https://github.com/alkem-io/alkemio/issues/758)  ✅ Fixed.
12. ~~Enable callout descriptions to consist of multiple paragraphs. Currently the editor supports this, but all is rendered as a single paragraph.~~  ✅ Fixed in https://github.com/alkem-io/client-web/issues/3131.
13. ~~Provide Unsplash search or other easily selectable cover photos for cards.~~  ✅ Not relevant anymore now that 10 is fixed.

Update to the changes requested in June 2023 by email (Re: chat follow up):

1. ~~Duplicate~~
2. **Clarify the Knowledge Base start page**, like at https://stackoverflow.com/questions ⏳ Still open.
3. **Merge Q&A into a clear integrated page** (1 question, multiple answers, small comments per Q/A), like at https://crypto.stackexchange.com/questions/53025/what-is-customizable-in-ecdsa-signature-and-verification ⏳ Still open.
4. **Ability to add instructions or limit options**: creation workflow for callout/answer. ⏳ Still open.
5. **Ability to add instructions**: top left in Knowledge Base. ⏳ Still open.
6. **Ability to link to a Q or A, without need to define the link title statically** ⚠️ This is still not fixed. This makes it hard for our users to reuse knowledge in their contributions.
7. ~~Counter for comments per card~~ ✅ Fixed.
8. ~~Duplicate~~
9. ~~Duplicate~~
10. ~~Duplicate~~
11. ~~Duplicate~~
12. **Ability to switch between NL and EN**, ideally with inline translation option

As discussed in [Choral Explanations](https://alkem.io/building-alkemio/collaboration/re-useofthealk-871/posts/choralexplanations-6752) on Building Alkemio, I’ve published a prototype for alternative rendering. Is it feasible to get some of these ideas into Alkemio?

**Comment by sander** [comment]
Update after meeting @SimoneZaza, in parallel with the Collaboration Tools UX feedback provided by @lvanderpeet.

- `2022-12-003` **Ability to edit ~~comments~~ <ins>callout answers</ins> while keeping version history.** ⏳ Detailed feedback shared, still awaiting commitment. Until then, we’ll limit edits to minor editorial issues.
- `2022-12-004` **Single-line title on a card preview often truncated too much to be useful.** ⏳ Discussed: making the title optional would be even better.
- `2022-12-007` **Ability to select an explicit Creative Commons license for a callout <ins>response</ins>.** ⚠️ Still painful. The proposed tagging solution will not help. Unclear if there is commitment to fix this.
- `2022-12-008` **Ability to upvote <ins>and downvote</ins> cards and comments**, such as on Stack Overflow. ⏳ The upcoming emoji feature may provide a useful first iteration. But anonymity is needed to provide useful upvotes and downvotes.
- `2023-06-002` **Clarify the Knowledge Base start page.** ⏳ Still open. Improving information density or guidance for first users is not planned yet.
- `2023-06-003` **Merge Q&A into a clear integrated page.** ⏳ A first step is planned to make it easier to navigate between question and answers. Feedback shared regarding better visual hierarchy.
- `2023-06-004` **Ability to add instructions or limit options.** ⏳ The template library is a useful first step, but we still miss a way to guide users to this instead of getting lost in one of the five other options.
- `2023-06-005` **Ability to add instructions top left in Knowledge Base.** ⏳ This feature may appear soon.
- `2023-06-006` **Ability to link to a Q or A, without need to define the link title statically** ⚠️ This is still not fixed nor planned. This makes it hard for our users to reuse knowledge in their contributions.
- `2023-06-012` **Ability to switch between NL and EN in contributions.** ⏳ This is not planned and will block further expansion of our community to non-NL contributors.

**Comment by larssondenise** [comment]
Closing as obsolete: comments and reactions ('liking') on callouts/contributions have already shipped — the domain has a dedicated reaction module. This 2020-era proposal is largely already implemented.

**Comment by larssondenise** [comment]
Merged into #1493, which has been rewritten as **Collective intelligence: Polis + Choral Explanations**.

Choral Explanations and Polis are two mechanisms for the same underlying question — how a community surfaces and learns from what it collectively knows — so they are now being discussed as one epic rather than two disconnected ones. #1493 sets out the distinction (anonymous/broad/statistical vs. attributed/deep/editorial) and carries the Choral Explanations scope forward in full.

Sub-issue #937 (Design for Choral explanations) has been re-parented to #1493. Please follow #1493 from here.
